docs/infra/treatment-service.md

```diff
@@ -0,0 +1,132 @@
+authentication, i.e the Pub/Sub topic and the BigQuery table each requiring a **different** service account for 
+authentication.
+
+### Deploying the Treatment Service
+
+#### As a Helm Release
+
+*Some of these steps have been adapted from https://github.com/caraml-dev/xp/tree/main/infra/charts/treatment-service*
+
+##### 1. Add the Helm Repository
+
+```sh
+$ helm repo add xp https://turing-ml.github.io/charts
+```
+
+##### 2. Install the Helm Chart
+
+This command will install XP Treatment Service release named `xp-treatment` in the `default` namespace.
+Default chart values will be used for the installation:
+```shell
+$ helm install xp-treatment xp/xp-treatment
+```
+
+You can (and most likely, should) override the default configuration with Helm chart values suitable for your 
+installation. Refer to [Configuration](https://github.com/caraml-dev/xp/tree/main/infra/charts/treatment-service#configuration) section for the detailed description of available configuration keys.
+
+You can also refer to [values.yaml](https://github.com/caraml-dev/xp/tree/main/infra/charts/treatment-service/values.yaml)
+for a minimal configuration that needs to be provided for XP Treatment Service installation.
+
+```shell
+$ helm install xp-treatment xp/xp-treatment \
+    --values=path/to/helm/chart/values/file.yaml
+```
+
+Notice that you can specify the *Treatment Service configuration* [values](#configuration) under the `xpTreatment` 
+field in the Helm chart values file like below: 
+
+```yaml
+xpTreatment:
+  config:
+    Port: 8080
+    ManagementService:
+      URL: https://caraml-dev.io/api/xp/v1
+      AuthorizationEnabled: true
+    ...
+```
+
+These configuration values would be saved in a `.yaml` file within a 
+[secret](https://kubernetes.io/docs/concepts/configuration/secret/) that gets mounted automatically onto the 
+Treatment Service pod, where it will be read by the Treatment Service.
+
+##### 2.1 Configure the Treatment Service to use a Google Cloud Provider (GCP) Service Account (Optional) 
+
+In particular, should you need to use a GCP service account for authenticating, one recommended way to set this up 
+is to utilise [volumes](https://kubernetes.io/docs/concepts/storage/volumes/) to mount the GCP service account key file, 
+which itself should be saved within a [secret](https://kubernetes.io/docs/concepts/configuration/secret/) in the 
+Kubernetes cluster, into the Treatment Service [pod](https://kubernetes.io/docs/concepts/workloads/pods/). Adding the 
+environment variable `GOOGLE_APPLICATION_CREDENTIALS` with the path to the mounted key file would then allow the 
+Treatment Service to access it.
+
+When deploying the Treatment Service with Helm, however, it is convenient to specify `extraVolumes`, 
+`extraVolumeMounts` and `extraEnvs` in the Helm chart values (see
+[values.yaml](https://github.com/caraml-dev/xp/tree/main/infra/charts/treatment-service/values.yaml#L59)), as they 
```

Comment:
Likewise here and everywhere else, `main` might make the links outdated over time.